src/amber/cli/templates/scaffold/controller/crecto/spec/controllers/{{name}}_controller_spec.cr.ecr

```diff
@@ -1,71 +1,86 @@
 <% field_name = @fields.first.name -%>
 require "./spec_helper"
 
-def create_subject
-  subject = <%= class_name %>.new
-  subject.<%= field_name %> = "test"
-  subject.save
-  subject
+def create_model
+  model = <%= class_name %>.new
+  <% if @fields.first.reference? -%>
+  model.<%= field_name %> = <%= field_name.camelcase %>.new 
+  <% else -%>
+  model.<%= field_name %> = "test"
+  <% end -%>
+  model.save
+  model
 end
 
-describe <%= class_name %>Controller do
+class <%= class_name %>ControllerTest < Spec::Controller::Test
```

Comment:
I'm getting an error, do we need to release a new version of `amber-spec` and have the generated app pick up that version? Or for testing tell it to pull from master, or a certain commit hash?
`undefined constant Spec::Controller::Test`
```
.====== START Creating database for test_app ======
====== DONE Database created test_app ======
FF.cleaning up...


Failures:

  1) building a generated app crystal spec can be executed
     Failure/Error: spec_result.should contain "Finished in"

       Expected:   "Error in line 1: while requiring \"./spec/controllers/animal_controller_spec.cr\"\n\nin spec/controllers/animal_controller_spec.cr:10: \e[1mundefined constant Spec::Controller::Test\e[0m\n\nclass AnimalControllerTest < Spec::Controller::Test\n                             \e[32;1m^~~~~~~~~~~~~~~~~~~~~~\e[0m\n\n"
       to include: "Finished in"

     # spec/amber/cli/build_spec.cr:47

  2) building a generated app crystal spec has no errors
     Failure/Error: spec_result.should_not contain "Error in line"

       Expected: value "Error in line 1: while requiring \"./spec/controllers/animal_controller_spec.cr\"\n\nin spec/controllers/animal_controller_spec.cr:10: \e[1mundefined constant Spec::Controller::Test\e[0m\n\nclass AnimalControllerTest < Spec::Controller::Test\n                             \e[32;1m^~~~~~~~~~~~~~~~~~~~~~\e[0m\n\n"
       to not include: "Error in line"

     # spec/amber/cli/build_spec.cr:51

Finished in 1:39 minutes
30 examples, 2 failures, 0 errors, 0 pending

Failed examples:

crystal spec spec/amber/cli/build_spec.cr:46 # building a generated app crystal spec can be executed
crystal spec spec/amber/cli/build_spec.cr:50 # building a generated app crystal spec has no errors
```